packages/scripts/src/migration/dvm2-designated-voting/0_Deploy.ts

```diff
@@ -0,0 +1,113 @@
+    designatedVotingContracts.map(async (address) => votingToken.balanceOf(address))
+  );
+  const designatedVotingContractsWithTokens: DesignatedVotingEthers[] = [];
+  for (let i = 0; i < designatedVotingContracts.length; i++)
+    if (contractsTokenBalances[i].gt(0))
+      designatedVotingContractsWithTokens.push(
+        await getContractInstance<DesignatedVotingEthers>("DesignatedVoting", designatedVotingContracts[i])
+      );
+
+  // Step 3: Find all owners and hot wallets for all DesignatedVoting contracts that have UMA. Construct one condensed
+  // data structure that contains all key information. Remove any that are not owned by the provided owner.
+  const owners = await Promise.all(designatedVotingContractsWithTokens.map((contract) => contract.getMember(0)));
+  const hotWallets = await Promise.all(designatedVotingContractsWithTokens.map((contract) => contract.getMember(1)));
+
+  const designatedVotingData = designatedVotingContractsWithTokens
+    .map((contract, index) => {
+      return { designatedVoting: contract.address, owner: owners[index], hotWallet: hotWallets[index] };
+    })
+    .filter((contract) => contract.owner === owner);
+
+  // Log all designated voting and the associated hot wallets. remove the owner element from the object to keep it short.
+  console.log(`Found the following DesignatedVoting to migrate owned by ${owner}:`);
+  console.table(
+    designatedVotingData.map((e: any) => {
+      delete e.owner;
+      return e;
+    })
+  );
+
+  // Step 4: Deploy new DesignatedVotingV2 contracts for each hot wallet.
+  const multiCallPayload = designatedVotingData.map(async (data) => {
+    await factoryV2.populateTransaction.newDesignatedVoting(owner, data.hotWallet);
+  });
+
+  console.log("multiCallPayload", multiCallPayload);
+}
+
+async function _fetchDesignatedVotingContractsCreatedByFactory(
+  factoryV1: DesignatedVotingFactoryEthers
+): Promise<string[]> {
+  const data = await _runEtherscanApiCall(
+    `https://api.etherscan.io/api?module=account&action=txlist&address=${factoryV1.address}&startblock=0&endblock=99999999&sort=asc`
+  );
+
+  const createCallers = data
+    .filter((tx: any) => tx.functionName.includes("newDesignatedVoting")) // choose only newDesignatedVoting calls.
+    .map((tx: any) => tx.from as string); // extract the caller address.
+
+  const designatedVotingFromCreation = await Promise.all(
+    createCallers.map((creator: string) => factoryV1.designatedVotingContracts(creator))
+  );
+
+  const setDesignatedVotingInputs = data
+    .filter((tx: any) => tx.functionName.includes("setDesignatedVoting")) // choose only setDesignatedVoting calls.
+    .map((tx: any) => tx.input as string); // extract the input calldata.
+
+  // Now, extract the target designated voting contract from the input calldata.
+  const designatedVotingFromSetDesignatedVoting = setDesignatedVotingInputs.map(
+    (input: string) => "0x" + input.substring(input.length - 40, input.length)
+  );
```

Comment:
This is a good solution. Annoying we had to do it this way -- I wish we had added events 🤦

Reply:
ye, a bit of a pain but a good technique to have in our arsenal.